src/lsm/table.zig

```diff
@@ -745,27 +773,52 @@ pub fn TableType(
             const data_block_index = binary_search.binary_search_keys_raw(
                 Key,
                 compare_keys,
-                Table.index_data_keys_used_const(index_block),
+                Table.index_data_keys_used(index_block),
                 key,
             );
             assert(data_block_index < index_data_blocks_used(index_block));
             return data_block_index;
         }
 
+        pub const IndexKeyInfo = struct {
+            filter_block_address: u64,
+            filter_block_checksum: u128,
+            data_block_address: u64,
+            data_block_checksum: u128,
+        };
+
+        /// Returns all data stored in the index block relating to a given key.
+        /// May be called on an index block only when the key is already in range of the table.
+        pub inline fn index_key_info(index_block: BlockPtrConst, key: Key) IndexKeyInfo {
```

Comment:
```suggestion
        pub inline fn index_blocks_for_key(index_block: BlockPtrConst, key: Key) IndexBlocks {
```

Or `index_filter_and_data_block_for_key() IndexFilterAndDataBlock`.